lib/Push.php

```diff
@@ -350,13 +419,21 @@ public function pushDeleteToDevice(string $userId, int $notificationId, string $
 			}
 
 			try {
-				$payload = json_encode($this->encryptAndSignDelete($userKey, $device, $notificationId));
-
 				$proxyServer = rtrim($device['proxyserver'], '/');
 				if (!isset($this->payloadsToSend[$proxyServer])) {
 					$this->payloadsToSend[$proxyServer] = [];
 				}
-				$this->payloadsToSend[$proxyServer][] = $payload;
+
+				if ($deleteAll) {
+					$data = $this->encryptAndSignDelete($userKey, $device, null);
+					$this->payloadsToSend[$proxyServer][] = json_encode($data['payload']);
+				} else {
+					while (!empty($notificationIds)) {
+						$data = $this->encryptAndSignDelete($userKey, $device, $notificationIds);
+						$notificationIds = $data['remaining'];
+						$this->payloadsToSend[$proxyServer][] = json_encode($data['payload']);
+					}
```

Comment:
This will only work for the first device, because for any device following, `$notificationIds` will be empty and not receive any notifications.

Reply:
Fixed